src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,233 @@
+*/
+record AnnotationSet {
+
+  /** The ID of the annotation set record */
+  string id;
+
+  /** The ID of the dataset to which this annotation set belongs */
+  string datasetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference datasets used.
+  */
+  Analysis analysis;
+
+}
+
+
+/**
+A transcript effect record is a set of information describing the
+effect of an allele on a transcript
+*/
+record TranscriptEffect {
+
+
+  /** The ID of the transcript effect record
+  */
+  string id;
+
+  //TODO: derive unique id from digest of data [location, allele, transcript?]
+
+
+  /** The id of the transcript feature the annotation is relative to
+  */
+  string featureId;
+
+
+  /** Alternate allele - a variant may have more than one alternate allele, each of which
+  will have distinct annotation. This is not be needed for the graph Allele model
+  */
+  union { null, string} alternateBases = null;
+
+  /** Effect of variant on this feature */
+  array<OntologyTerm> effects;
+
+
+  /** Highest Impact from the predicted effects */
+  Impact impact;
+
+
+  /** HGVS formatted annotation at genomic level  */
+  union { null, string } HGVSg = null;
+
+  /** HGVS formatted annotation at transcript level */
+  union { null, string } HGVSc = null;
+
+  /** HGVS formatted annotation at protein level */
+  union { null, string } HGVSp = null;
+
+
+  /** Change relative to cDNA  */
+  union {null, AlleleLocation} cDNALocation =null;
+
+  /* Change relative to coding sequence  */
+  union { null, AlleleLocation} CDSLocation = null;
+
+  /** Change relative to protein  */
+  union { null, AlleleLocation} proteinLocation = null;
+
+
+  /** Output from prediction packages such as Sift */
+  array<AnalysisResult> analysisResults;
+
+
+}
+
+
+
+
+/**
+A `VariantAnnotation` record represents the result of comparing a variant
+to a set of reference data.
+*/
+record VariantAnnotation {
+
+  /** The variant ID. */
+  string id;
+
+  /** The ID of the annotation set this record belongs to. */
+  string annotationSetId;
+
+
+  /** The date this annotation was created in milliseconds from the epoch. */
+  union { null, long } created = null;
+
+
+  /**
+  The transcript effect annotation for the alleles of this variant. Each one
+  represents the effect of a single allele on a single transcript.
+  */
+  array<TranscriptEffect> transcriptEffects = [];
+
+
+  /**
+  The names of known variants which are co-located with this variant.
+  these can use used to look up disease associations, ClinVar statuses,
+  allele frequencies in reference panels, etc
+  */
+  array<string> coLocatedVariants =[];
+
+
+  /** Additional annotation data in key-value pairs. */
+  map<array<string>> info = {};
+
+}
+
+/**
+An `AlleleAnnotation` record represents the result of comparing a allele
+to a set of reference data.
+*/
+record AlleleAnnotation {
+
+  /** The allele ID. */
```

Comment:
I've added ids for both variant and allele annotation records - these were not present initially.